### PR TITLE
created solution

### PR DESCRIPTION
**Pull request by d1kenpiken**

### CODE REVIEW

File: src/main/java/core/basesyntax/Application.java [hunk 1]
```diff
@@ -2,6 +2,13 @@
...
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+
+        Ball firstBall = Lottery.getRandomBall();
+        Ball secondBall = Lottery.getRandomBall();
+        Ball thirdBall = Lottery.getRandomBall();
+
+        System.out.println(firstBall);
+        System.out.println(secondBall);
+        System.out.println(thirdBall);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Application.java:12]: can you somehow avoid code duplication here and make it shorter/simpler? Consider using loop for your purposes

File: src/main/java/core/basesyntax/Lottery.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import java.util.Random;
+
+public class Lottery {
+    public static Ball getRandomBall() {
+
+        Color randomColor = ColorSupplier.getRandomColor();
+        Random random = new Random();
+        int randomNumber = random.nextInt(100) + 1;
+        return new Ball(randomColor, randomNumber);
+    }
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:12]: let's not make this method static if there is no crucial need

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public static Ball getRandomBall() {
+
+        Color randomColor = ColorSupplier.getRandomColor();
+        Random random = new Random();
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:9]: new Random object is created each time the method is called. Let's move it to the class fields

File: src/main/java/core/basesyntax/Lottery.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public static Ball getRandomBall() {
+
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:7]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+        Color randomColor = ColorSupplier.getRandomColor();
+        Random random = new Random();
+        int randomNumber = random.nextInt(100) + 1;
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:10]: magic number

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 6]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static Color getRandomColor() {
+
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: the same problem with static method here. And redundant empty line

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 7]
```diff
@@ -1,7 +1,12 @@
...
+        Color[] colors = Color.values();
+        Random random = new Random();
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/ColorSupplier.java:9]: same with random

**Review by sarakhmen — CHANGES_REQUESTED** [review]
Let's improve your solution :)

File: src/main/java/core/basesyntax/Application.java (hunk 1 above)
Comment (MentorMA) [reply src/main/java/core/basesyntax/Application.java:12]: https://github.com/mate-academy/jv-lottery/blob/master/checklist.md read this at first

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 8]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static Color getRandomColor() {
```
Comment (MentorMA) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: Return color in String format

**Review by MentorMA — CHANGES_REQUESTED** [review]
Fix please previous mentor`s comments

File: src/main/java/core/basesyntax/Application.java (hunk 1 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/Application.java:12]: not fixed

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 8 above)
Comment (kerrrusha) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: should not be static
```suggestion
    public Color getRandomColor() {
```

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 7 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/ColorSupplier.java:9]: not fixed

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 6 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/ColorSupplier.java:7]: not fixed

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 8 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/ColorSupplier.java:6]: not fixed

File: src/main/java/core/basesyntax/Lottery.java (hunk 4 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/Lottery.java:7]: not fixed

File: src/main/java/core/basesyntax/Lottery.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public static Ball getRandomBall() {
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Lottery.java:6]:
```suggestion
    public Ball getRandomBall() {
```

File: src/main/java/core/basesyntax/Lottery.java (hunk 3 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/Lottery.java:9]: not fixed

File: src/main/java/core/basesyntax/Lottery.java (hunk 5 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/Lottery.java:10]: not fixed

File: src/main/java/core/basesyntax/Lottery.java (hunk 2 above)
Comment (kerrrusha) [reply src/main/java/core/basesyntax/Lottery.java:12]: not fixed

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Ball.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class Ball {
+    private final String color; // Changed to String
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Ball.java:4]: remove redundant comment ;)

File: src/main/java/core/basesyntax/Lottery.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_NUMBER = 100;
+    private static final Random random = new Random();
+
+    public Ball getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier();
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:10]: please move ColorSupplier to the fields, likewise, you did with random

File: src/main/java/core/basesyntax/Lottery.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+        String randomColor = colorSupplier.getRandomColor(); // Now this should return a String
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Lottery.java:11]:
```suggestion
        String randomColor = colorSupplier.getRandomColor();
```

**Review by sarakhmen — CHANGES_REQUESTED** [review]

**Pull request closed without merging**